src/weather.cpp

```diff
@@ -463,13 +461,10 @@ void weather_effect::thunder()
         if( g->get_levz() >= 0 ) {
             add_msg( _( "You hear a distant rumble of thunder." ) );
             sfx::play_variant_sound( "environment", "thunder_far", 80, rng( 0, 359 ) );
-        } else if( g->u.has_trait( trait_GOODHEARING ) && one_in( 1 - 2 * g->get_levz() ) ) {
+        }  else if( one_in( std::max( round( 2 * g->get_levz() / Character::mutation_value( "hearing_modifier" ) ) , 1  ) ) ) {
             add_msg( _( "You hear a rumble of thunder from above." ) );
```

Comment:
This is much better then my initial proposal.
But using `round` will return same results for all modifiers in the range from 1 to 1.49, it will be better to use `roll_remainder` which will randomly return integer number based on probability. And replace integer `2` with `2.0f` otherwise argument for the round function will always be integer afaik.
So this line will be
```
} else if( one_in( std::max( roll_remainder( 2.0f * g->get_levz() / Character::mutation_value( "hearing_modifier" ) ), 1 ) ) ) { {
```